core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.js

```diff
@@ -0,0 +1,691 @@
+const millionsOfContentId = '#about-foundation-page-millions-of-content';
+const weCannotContentId = '#about-foundation-page-we-cannot-content';
+const sourceUnescoButtion = 'a#about-foundation-page-source-unesco-button';
+const learnMoreAboutOppiaButton =
+  '#about-foundation-page-learn-more-about-oppia-button';
+const becomeAVolunteerButton =
+  '#about-foundation-page-become-a-volunteer-button';
+const sectionSixPart1 = '#about-foundation-page-section-six-part-1';
+const sectionSixPart2 = '#about-foundation-page-section-six-part-2';
+const sectionSixPart3 = '#about-foundation-page-section-six-part-3';
+
+const watchAVideoButton = 'a#thanks-for-donating-page-watch-a-video-button';
+const readOurBlogButton = 'a#thanks-for-donating-page-read-our-blog-button';
+const dismissButton = 'i#thanks-for-donating-page-dismiss-button';
+const thanksForDonatingClass = '.modal-open';
+const donatePage = '.modal-backdrop.fade';
+const thanksForDonatingPage = '.modal-backdrop.fade.show';
+
+module.exports = class LoggedInUsers extends baseUser {
+  /**
+   * Function for navigating to the home page.
+   */
+  async navigateToHome() {
+    await this.goto(homeUrl);
+  }
+
+  /**
+   * Function for navigating to the about page.
+   */
+  async navigateToAboutPage() {
+    await this.goto(aboutUrl);
+  }
+
+  /**
+   * Function for navigating to the about foundation page.
+   */
+  async navigateToAboutFoundationPage() {
+    await this.goto(aboutFoundationUrl);
+  }
+
+  /**
+   * Function for navigating to the Thanks for Donating page.
+   */
+  async navigateToThanksForDonatingPage() {
+    await this.page.goto(thanksForDonatingUrl, {
+      waitUntil: 'load',
+      timeout: 0
+    });
+  }
+
+  /**
+   * Function to click the About button in the About Menu on navbar
+   * and check if it opens the About page.
+   */
+  async clickAboutInAboutMenuOnNavbar() {
+    await this.clickOn(navbarAboutTab);
+    await this.clickOn(navbarAboutTabAboutButton);
+    if (this.page.url() !== aboutUrl) {
+      throw new Error (
+        'The About Oppia button in About Menu on navbar ' +
+        'does not open the About page!');
+    } else {
+      showMessage(
+        'The About Oppia button in About Menu on navbar ' +
+        'opens the About page.');
+    }
+  }
+
+  /**
+   * Function for click the Browse Our Lessons button in the About page
+   * and check if it opens the Math Classroom page.
+   */
+  async clickBrowseOurLessonsButtonInAbout() {
+    await this.clickOn(browseOurLessonsButton);
+    if (this.page.url() !== mathClassroomUrl) {
+      throw new Error (
+        'The Browse our lesson button does not open the Math Classroom page!');
+    } else {
+      showMessage(
+        'The Browse our lesson button opens the Math Classroom page.');
+    }
+  }
+
+  /**
+   * Function for click the Access Android App button in the About page
+   * and check if it opens the Android page.
+   */
+  async clickAccessAndroidAppButtonInAbout() {
```

Comment:
There's a fair amount of repetition here and I'm wondering if we need to repeat the same sort of logic elsewhere. How about we have a private function here that does something like this:

```
private async clickButtonToNavigateToNewPage(button, buttonName, expectedDestinationPageUrl, expectedDestinationPageName)
```

(or you can use the '#' syntax described later in https://www.typescriptlang.org/docs/handbook/2/classes.html#private )

and then the functions here can use it, e.g. clickButtonToNavigateToNewPage(accessAndroidAppButton, 'Access the Android App', androidUrl, 'Android')

For anything that doesn't follow the standard template, it's fine to keep the longer impl, but what do you think about using this to avoid all if/else repetition? That would make it easier to add new actions too.